### ISSUE TITLE
Explore different models for training

### ISSUE THREAD

**Issue by sunraymoonbeam** [shafiqninaba/resonare#29]

**Comment by sunraymoonbeam** [comment]
model_map = {
                "unsloth/gemma-3-4b-it-unsloth-bnb-4bit": "gemma-3",
                "unsloth/Llama-3.2-3B-Instruct-bnb-4bit": "llama-3.1",
                "unsloth/Meta-Llama-3.1-8B-Instruct-bnb-4bit": "llama-3.1",
            }

**Comment by sunraymoonbeam** [comment]
![Image](https://github.com/user-attachments/assets/e3fa3d3d-4d04-4ed5-9d89-0606da3ed98f)

Model seems pretty smart but not trained?? perhaps not enough data?

**Comment by sunraymoonbeam** [comment]
Update: unsloth/gemma-3-1b-it-unsloth-bnb-4bit works for training, but inference is wonky.

unsloth-backend-1  | {'loss': 3.6169, 'grad_norm': 1.6641143560409546, 'learning_rate': 0.00018461538461538463, 'epoch': 0.25}
unsloth-backend-1  | {'loss': 4.6243, 'grad_norm': 1.7131824493408203, 'learning_rate': 0.00017692307692307693, 'epoch': 0.28}
unsloth-backend-1  | {'loss': 4.313, 'grad_norm': 1.8258556127548218, 'learning_rate': 0.00016923076923076923, 'epoch': 0.31}

**Comment by sunraymoonbeam** [comment]
experiment with different instruction tuned models like gemma, or mistral.

standard chat templates such as chatml could work as well with base models, but for now it seems that instruction tuned models are superior
